test/lib/mock-dbus.ts

```diff
@@ -0,0 +1,48 @@
+import * as dbus from 'dbus';
+import { DBusError, DBusInterface, busType, DBusConnection } from 'dbus';
+import { createSandbox, SinonSandbox, SinonStub } from 'sinon';
+
+import * as dbusLib from '../../src/lib/dbus';
+
+let sandbox: SinonSandbox;
+
+const dbusLibFunctions = Object.keys(dbusLib).filter((k) => k !== 'DbusError');
+
+const stubs: { [key: string]: SinonStub } = {};
+
+export const mock = () => {
+	console.log('Mocking dbus...');
+
+	sandbox = createSandbox();
+	sandbox.stub(dbus, 'getBus').callsFake((_type: busType) => {
+		return {
+			getInterface(
+				_serviceName: string,
+				_objectPath: string,
+				_interfaceName: string,
+				_callback: (err: DBusError, iface: DBusInterface) => void,
+			) {
+				// noop
+			},
+		} as DBusConnection;
+	});
+
+	dbusLibFunctions.forEach((fn) => {
+		stubs[fn] = sandbox.stub(dbusLib, fn as any);
+	});
+};
+
+export const getStub = (method: string) => {
```

Comment:
I understand why we are chosing to stub methods in the dbus library, but it would really be nicer imho if we could actually mock responses coming from the dbus service, avoiding to rely on internal implementation checks.

For instance something like 

```
expect(restartService.callCount).to.equal(2);
```

could be replaced by an integration test that checks that the right call was made to the dbus service. 

This is definitely not the highest priority right now, but is something to keep in mind.